app/estimate_square_size.go

```diff
@@ -170,7 +170,7 @@ func rawShareCount(txs []*parsedTx, evd core.EvidenceList) (txShares, evdShares
 		// compensates for the actual size of the message, and in some cases can
 		// result in some wasted square space or picking a square size that is
 		// too large. TODO: improve by making a more accurate estimation formula
-		txBytes += overEstimateMalleatedTxSize(len(pTx.rawTx), len(pTx.msg.Message), len(pTx.msg.MessageShareCommitment))
+		txBytes += overEstimateMalleatedTxSize(len(pTx.rawTx), len(pTx.msg.Message), 1)
```

Comment:
[Blocking]
```
// overEstimateMalleatedTxSize estimates the size of a malleated tx. The formula it uses will always over estimate.
func overEstimateMalleatedTxSize(txLen, msgLen, sharesCommitments int) int {
	// the malleated tx uses the original txLen to account for meta data from
	// the original tx, but removes the message
	malleatedTxLen := txLen - msgLen
	// we need to ensure that the returned number is at least larger than or
	// equal to the actual number, which is difficult to calculate without
	// actually malleating the tx
	return appconsts.MalleatedTxBytes + appconsts.MalleatedTxEstimateBuffer + malleatedTxLen
}
```
The third argument doesn't seem to be used, why keep it?

Reply:
Great catch, will remove!

Reply:
resolved in https://github.com/celestiaorg/celestia-app/pull/937/commits/b7aa8936b6a1925a662130c8387dd4868c938c9c